### ISSUE TITLE
tab key in data editor value box changes focus

### ISSUE THREAD

**Issue by phcoffin** [dbeaver/dbeaver#22318]
### Description

Use of Tab key in data value pane to enter a tab character into a text column refocuses to the data grid pane, and subsequent typing overwrites text previously entered in the data value pane.

### DBeaver Version

Community Edition 23.3.1

### Operating System

Sonoma 14.2.1

### Database and driver

Editor problem, crosses databases, verified in 

sqlite_jdbc
IBM Db2 for Linux/Unix/Windows

### Steps to reproduce

1. Connect to any databse
2. Open any table for data editing
3. Select any text column row entry
4. Enter any text in the Value pane including pressing the tab key to enter a tab character such as "0900[tab]visit garden"
5. Instead of accepting the text as entered, DBeaver will accept "0900[tab]" then refocus to the grid pane and overwrite the previous text with the rest of the typing ("visit garden")
6. clicking the value pane after the refocus but before any other typing will return focus to the value pane, showing the tab character correctly entered, and allow completion of the typing, until either the column cell is moved off of (saving the value for subsequent commit) or the next tab keystroke happens.

### Additional context

Same wrong behavior exists on Version 23.3.1.202312241705 running on Windows 11 Pro Version 23H2

**Comment by E1izabeth** [comment]
I was able to reproduce focus losing, but for me whole text value from value panel is in the cell, including tab
![image](https://github.com/dbeaver/dbeaver/assets/28875055/458b52a6-d134-4360-9dea-d4a99debea3e)

**Comment by phcoffin** [comment]
Right, it's still entered, _until_ the next character is typed in the refocused data cell location, which clears the entry, replacing it with the typed character. If the first action after hitting [tab] is to refocus back to the Value pane, it's fine, plus or minus having to take that extra action.

**Comment by E1izabeth** [comment]
I see. Thank you for bug report.

**Comment by phcoffin** [comment]
Adding: this focus change only happens on the TAB keystroke. Using ctrl-I to enter a TAB character enters the TAB character as expected and does not refocus.
